api/src/routes/users.js

```diff
@@ -131,4 +132,19 @@ router.delete(
   }),
 );
 
+// add new route to hard delete a user
+router.delete(
+  '/:username/delete',
+  isPermittedTo('delete', { checkOwnerShip: true }),
```

Comment:
Can you use a single DELETE endpoint for both hard and soft-delete cases? Try to reuse the existing one.

To reuse, send a query parameter called `hard_delete` in the URL (in the existing DELETE endpoint). Default it to `false`. Send the query param as `true` if you want to hard-delete. Then, within the endpoint definition, check the value of `req.query.hard_delete` to determine if the endpoint needs to do a soft-deletion or a hard-deletion.

Examples of boolean query params: https://github.com/IUSCA/bioloop/blob/main/api/src/routes/datasets.js#L284

Reply:
Also, please ensure (if you haven't already) that any new routes you add are idempotent. That is, the end result of the operation (for example, database updates done by the route) stays the same, no matter how many times the endpoint is called.